File: src/main/java/jPetrovich/util/rules/RulesLoader.java
/*$Id:$*/
package jPetrovich.util.rules;

import jPetrovich.util.rules.data.Rules;
import org.ho.yaml.Yaml;

import java.io.File;
import java.io.FileNotFoundException;

/**
 * @author DMITRY KNYAZEV
 * @since 31.05.2014
 */
public final class RulesLoader {
	private RulesLoader() {
		throw new AssertionError("It helper class. Create instance");
	}

	/**
	 * Load rules from JAML file by default path
	 *
	 * @return rules
	 * @throws FileNotFoundException if file path not exist
	 */
	public static Rules loadRules() throws FileNotFoundException {
		return loadRules("rules.yml");
	}

	/**
	 * Load rules from JAML file path
	 *
	 * @param path path to rules file
	 * @return rules
	 * @throws FileNotFoundException if file path not exist
	 */
	public static Rules loadRules(String path) throws FileNotFoundException {
		return Yaml.loadType(new File(path), Rules.class);
	}
}

File: src/main/kotlin/kt/petrovich/rules/RulesLoader.kt
package kt.petrovich.rules

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
import java.io.File
import java.io.FileNotFoundException

/**
 * @author Dmitrii Kniazev
 * @since 31.05.2014
 */
class RulesLoader private constructor() {
    companion object {
        private val defaultRulesPath = Thread.currentThread()
                .contextClassLoader
                .getResource("rules.yml").path

        /**
         * Load rules from YAML file path
         *
         * @param path path to rules file
         * @return rules
         * @throws FileNotFoundException if file path not exist
         */
        @Throws(FileNotFoundException::class)
        @JvmOverloads
        fun loadRules(path: String = defaultRulesPath): Rules {
            return ObjectMapper(YAMLFactory()).readValue(File(path), Rules::class.java)
        }
    }
}
